Library/Homebrew/messages.rb

```diff
@@ -0,0 +1,36 @@
+# A Messages object collects messages that may need to be displayed together
+# at the end of a multi-step `brew` command run
+class Messages
+  attr_reader :caveats, :formula_count
+
+  def initialize
+    @caveats = []
+    @formula_count = 0
+  end
+
+  def record_caveats(f, caveats)
+    @caveats.push([f, caveats])
```

Comment:
A hash might be nicer here?

Reply:
Not sure a hash would be best — I'd like to preserve the ordering of the messages wrt how they were first encountered. (I'm thinking ahead to build times here.) Or maybe alphabetical would be better?

Reply:
`@caveats.push({f.name => caveats})` perhaps?

Reply:
That gets kind of ugly when doing the reading:

```
  ohai c.keys.first, c[c.keys.first]
```

Maybe `@caveats.push({"name" => f.name, "caveats" => caveats})`?

Reply:
Yeh, that looks nicer. I'd still be tempted to just store the formula name, though, if that's all you're using?

Reply:
Amended to store just the formula name.